lib/metasploit/framework/credential_collection.rb

```diff
@@ -1,233 +1,292 @@
           end
-          if pass_fd
-            pass_fd.each_line do |pass_from_file|
-              pass_from_file.chomp!
-              yield Metasploit::Framework::Credential.new(public: user_from_file, private: pass_from_file, realm: realm, private_type: private_type(pass_from_file))
+          pass_fd.seek(0)
+        end
+        additional_privates.each do |add_private|
+          yield Metasploit::Framework::Credential.new(public: username, private: add_private, realm: realm, private_type: private_type(add_private))
+        end
+      end
+
+      if user_file.present?
+        File.open(user_file, 'r:binary') do |user_fd|
+          user_fd.each_line do |user_from_file|
+            user_from_file.chomp!
+            if password.present?
+              yield Metasploit::Framework::Credential.new(public: user_from_file, private: password, realm: realm, private_type: private_type(password) )
+            end
+            if user_as_pass
+              yield Metasploit::Framework::Credential.new(public: user_from_file, private: user_from_file, realm: realm, private_type: :password)
+            end
+            if blank_passwords
+              yield Metasploit::Framework::Credential.new(public: user_from_file, private: "", realm: realm, private_type: :password)
+            end
+            if pass_fd
+              pass_fd.each_line do |pass_from_file|
+                pass_from_file.chomp!
+                yield Metasploit::Framework::Credential.new(public: user_from_file, private: pass_from_file, realm: realm, private_type: private_type(pass_from_file))
+              end
+              pass_fd.seek(0)
+            end
+            additional_privates.each do |add_private|
+              yield Metasploit::Framework::Credential.new(public: user_from_file, private: add_private, realm: realm, private_type: private_type(add_private))
             end
-            pass_fd.seek(0)
-          end
-          additional_privates.each do |add_private|
-            yield Metasploit::Framework::Credential.new(public: user_from_file, private: add_private, realm: realm, private_type: private_type(add_private))
           end
         end
       end
-    end
 
-    if userpass_file.present?
-      File.open(userpass_file, 'r:binary') do |userpass_fd|
-        userpass_fd.each_line do |line|
-          user, pass = line.split(" ", 2)
-          if pass.blank?
-            pass = ''
-          else
-            pass.chomp!
+      if userpass_file.present?
+        File.open(userpass_file, 'r:binary') do |userpass_fd|
+          userpass_fd.each_line do |line|
+            user, pass = line.split(" ", 2)
+            if pass.blank?
+              pass = ''
+            else
+              pass.chomp!
+            end
+            yield Metasploit::Framework::Credential.new(public: user, private: pass, realm: realm)
```

Comment:
Shouldn't we also technically execute `chomp!` on `user` here as well?

Reply:
No because in the file format the user and password values are on the same line so only the password will have a newline which is what `chomp` is there to remove.

Reply:
My apologies I mistakenly thought `chomp` was going to remove any additional trailing spaces which was try I originally suggested it. Appears I confused it with a different method.